File: catkin_ws/src/.old_archive/learning_topic/script/person_publisher.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import rospy
from learning_topic.msg import Person

def person_publisher():
	# node init
	rospy.init_node('person_publisher',anonymous=True)
	
	# create publisher
		# Topic name: /person_info
		# msg type: learning_topic::Person
		# line length: 10
	person_info_pub = rospy.Publisher('/person_info',Person,queue_size=10)
	
	# set loop freq
	rate=rospy.Rate(10)
	
	while not rospu.is_shutdown():
		# init message of this (geometry_msgs::Twist) type
		person_msg=Person()
		person_msg="Tom"
		person_msg=18
		person_msg.sex=Person.male
		
		# publish message
		person_info_pub.publish(person_msg)
		rospy.loginfo("Publish person message[%s, %d, %d]",
				person_msg.name, person_msg.age, person_msg.sex)
				
		# delay for set time loop
		rate.sleep()
		
if __name__=='__main__':
	try:
		person_publisher()
	except rospy.ROSInterruptException:
		pass

File: catkin_ws/src/.old_archive/learning_topic/src/person_publisher.cpp
// publish the topic: /person_info
// user defined message type: learning_topic::Person
#include <ros/ros.h>
#include "learning_topic/Person.h"

int main (int argc, char **argv)
{
	// node init. Node name: velocity_publisher
	ros::init(argc, argv, "person_publisher");
	
	// create node handle
	ros::NodeHandle n;
	
	// Create publisher. 
	// Topic name: /person_info
	// message type: learning_topic::Person
	// line length=10
	ros::Publisher person_info_pub = n.advertise<learning_topic::Person>("/person_info",10);
	
	// set loop freq
	ros::Rate loop_rate(1);
	
	int count=0;
	while (ros::ok())
	{
		// init geometry_msgs::Twist message
		learning_topic::Person person_msg;
		person_msg.name="Tom";
		person_msg.age=18;
		person_msg.sex=learning_topic::Person::male;
		
		// publish message
		person_info_pub.publish(person_msg);
		ROS_INFO("Publish person info: Name:%s  Age:%d  Sex:%d",
			person_msg.name.c_str(),person_msg.age, person_msg.sex);
			
		// delay for set time loop
		loop_rate.sleep();
	}
	
	return 0;
	
}
